rollout/trafficrouting/nginx/nginx.go

```diff
@@ -124,7 +124,8 @@ func (r *Reconciler) buildCanaryIngress(stableIngress *networkingv1.Ingress, nam
 
 	// Process additional annotations, would commonly be things like `canary-by-header` or `load-balance`
 	for k, v := range r.cfg.Rollout.Spec.Strategy.Canary.TrafficRouting.Nginx.AdditionalIngressAnnotations {
-		if !strings.HasPrefix(k, annotationPrefix) {
+		// If another slash is detected, assume the user has provided the full annotation key, and, therefore, do not prefix it
```

Comment:
Rather than assuming, could we explicitly get this "intent" from the user through config?

Reply:
We should add another config called `annotation` alongside `annotationPrefix`, add validation(s) to only have one of the two specified and document them so that the users know when to use which.

Reply:
Just to make sure I'm not misinterpreting, what would be the value of `annotation`? Would it be a key-value map of custom, full annotations to append to the object?

Reply:
It would be the full annotation (without the prefix). If it's more clear you can name it `fullAnnotation` and the explain in the docs what's the difference between this and the `annotationPrefix`

I'd also let @zachaller take a look in case he has a preference.

Reply:
It's a bit hard because I don't like the current API design at all. We basicly want to give end users the option to add annotations to the argo rollouts managed canary ingress. I do like the idea of trying to capture that intent I just don't love the options all that much.

Some ideas:
```
          annotationPrefix: customingress.nginx.ingress.kubernetes.io # optional
          additionalIngressAnnotations:   # optional
            canary-by-header: X-Canary
            canary-by-header-value: iwantsit
            my-custom-annotation.mygroup.com/key: value
```
```
          annotationPrefix: customingress.nginx.ingress.kubernetes.io # optional
          additionalIngressAnnotations:   # optional
            canary-by-header: X-Canary
            canary-by-header-value: iwantsit
          additionalCanaryIngressAnnotations:
            my-custom-annotation.mygroup.com/key: value
```
```
          annotationPrefix: customingress.nginx.ingress.kubernetes.io # optional
          additionalIngressAnnotations:   # optional
            canary-by-header: X-Canary
            canary-by-header-value: iwantsit
          customCanaryIngressAnnotations:
            my-custom-annotation.mygroup.com/key: value
```
```
          annotationPrefix: customingress.nginx.ingress.kubernetes.io # optional
          additionalIngressAnnotations:   # optional
            canary-by-header: X-Canary
            canary-by-header-value: iwantsit
          canaryIngressAnnotations:
            my-custom-annotation.mygroup.com/key: value
```

I think `canaryIngressAnnotations` might be the cleanest option, and would let us also add something like `canaryIngressLabels`

Reply:
I agree that it is the cleanest, and would also more easily allow for a future transition into using solely that field for extra annotations (eventually deprecating `annotationPrefix` + `additionalIngressAnnotations` in favor of it - better for users to be clearer). If we're agreed, I'll modify the proposal to accommodate that

Reply:
after a struggle with conflicts (sorry about all the opening/opening shenanigans, messy rebasing), I believe I've refactored accordingly. Please take another look whenever you can @zachaller @agrawroh